src/Veldrid/OpenGL/NoAllocEntryList/NoAllocUpdateBufferEntry.cs

```diff
@@ -4,10 +4,10 @@ internal struct NoAllocUpdateBufferEntry
     {
         public readonly Tracked<DeviceBuffer> Buffer;
         public readonly uint BufferOffsetInBytes;
-        public readonly Tracked<byte[]> StagingBlock;
+        public readonly StagingBlock StagingBlock;
```

Comment:
There's a subtle problem with this -- `StagingBlock` contains a managed reference (to the pool), but `NoAllocUpdateBufferEntry` instances are stored in a non-GC-tracked unsafe buffer. This could potentially introduce GC holes, although it's very unlikely since the lifetime of these entries should be strictly within the lifetime of the GraphicsDevice anyways.

To fix this, we should remove the StagingMemoryPool reference from StagingBlock, and pass the pool in to the Free method.